src/audio/dai-zephyr.c

```diff
@@ -1141,8 +1141,12 @@ static int dai_comp_trigger_internal(struct dai_data *dd, struct comp_dev *dev,
 		 * leading to the DMA copying stale data due to
 		 * dma_status() stopping dai_common_copy() from
 		 * updating the data.
+		 *
+		 * Only applies to non HD-DMA links as HD-DMA read/write pointer
+		 * is not reset during stop/config/start
 		 */
-		audio_stream_reset(&dd->dma_buffer->stream);
+		if (!(dd->dai->dma_caps & DMA_CAP_HDA))
+			audio_stream_reset(&dd->dma_buffer->stream);
```

Comment:
@LaurentiuM1234 @dbaluta With this fixup in place, this does not look like a solid design anymore. Either SOF can assume something about Zephyr DMA driver behaviour, or it cannot. It does sound like instead of audio_stream_reset, SOF should synchronize its view of DMA driver state (either at start or stop) and we could again have code that works for all (then it's upto DMA drivers how they behave for pause/reset).

Not blocking this PR as this fixes a rather nasty bug with Intel HDA devices, but I think this needs further work. The HW details should be handled on Zephyr side and SOF code should be agnostic.

Reply:
ACK. Just curious, why do we need the `dma_stop()` and `dma_config()` operations on release? If I recall correctly, an alternative to https://github.com/thesofproject/sof/commit/9831a9ded7701416f5eab1ab745383041bdb42dc was to just, well, not do those, which worked on NXP platforms at least? This would have been another way of keeping the buffers consistent with each other (i.e: SOF view and DMA view)

Reply:
@LaurentiuM1234, HD-DMA zephyr driver does not support dma_resume(), other than that, I don't know why we do the stop/start here.. We do clear the buffer, so it would kind of make sense to stop/start the DMA, but why we clear the buffer on pause/resume? I don't know.